### ISSUE TITLE
Apache Image - islandora user UID / GUID should be 10000 (v1.0) not 1000 currently

### ISSUE THREAD

**Issue by g7morris** [Islandora-Collaboration-Group/ISLE#187]
#### Issue description

On the Apache Image the islandora user UID / GUID is currently set to 1000 which causes issues when attempting to use bind mounts on systems that have a preexisting user that is not islandora with a 1000 UID / GUID. In essence the islandora user on the host server doesn't match the islandora on the apache container and websites won't work properly.  

This will impact all users of external new and migrated ISLE sites. 

Recommend rebuilding Apache image with the islandora user UID / GUIDs set to 10000 please.

#### Steps to reproduce the issue

1.  Review the ISLE v1.0 setting: https://github.com/Islandora-Collaboration-Group/ISLE/blob/ISLE-v1.0/apache/Dockerfile 

Lines 139 & 140
```
 && groupadd -g 10000 islandora \
    && useradd -m -d /home/islandora -s /bin/bash -u 10000 -g 10000 islandora \ 
```

2. Review the ISLE v1.0 setting for the Ansible hostvars deploy code: https://github.com/Islandora-Collaboration-Group/ISLE/blob/ISLE-v1.0/ansible/host_vars/host_local_macos_isle_localdomain.yml

Lines 16 & 17 
```
group_id_number: 10000
user_id_number: 10000
```

3. Compare to current Apache Image Dockerfile settings. https://github.com/Islandora-Collaboration-Group/isle-apache/blob/7be53f841b8b8de2f202c60c19a0b6cb3b867ede/Dockerfile

Line 4

```
    ISLANDORA_UID=${ISLANDORA_UID:-1000} \
``` 

#### What's the expected result?

- That running scripts that fix / set permissions (e.g. fix-permissions.sh) on Apache websites '/var/www/html/`  display the correct user islandora with UID and GUID of 10000

#### What's the actual result?

- If using an AWS Ubuntu 18.04 host server, the first created user is `ubuntu` that has a UID / GUID of 1000 and as such the secondary `islandora` user account with 10000 (created from Ansible deploy) doesn't match the Apache container's `islandora` user with the the UID / GUID of 1000.

Thus the permissions deny the host `islandora` account user access to bind mounted volumes that contain Drupal code and as such to the outside world.

**Comment by br2490** [comment]
This was intentional. `1000` was used because for most systems this is the first "real user" UID. 
The UID of Islandora should match the UID of the user who will interact with the webroot. This mostly deprecates the need to create an islandora user on the host machine.  
- The environment variable of ISLANDORA_UID can be set in the .env or in the environment section of a docker-compose.yml to specify the appropriate numerical id of the user.  
- [ ] Please verify that  `ISLANDORA_UID=10000` will achieve the desired result of setting that UID to the files in the webroot!

**Comment by br2490** [comment]
Oh this makes sense - never implemented in the cont-init.d adding this for
next set of updates.

Benjamin Rosner

Instructional Application Developer
Instructional Media and Technology,
Barnard Library and Academic Information Services
p: 212-854-9005

**Comment by br2490** [comment]
Image pushed to Dockerhub as isle-apache:1.1 (default).  Please docker-pull and attempt using ENV to set UID! 
Report issues here!
This will be incorporated into a large set of changes for publication to the Google Group - testing and feedback please before then, please @g7morris @Islandora-Collaboration-Group/isle-maintainers-commiters-contributors!

**Comment by g7morris** [comment]
@br2490 we should change the current `.env` to have the `ISLANDORA_UID=10000` setting.

**Comment by g7morris** [comment]
@br2490 I can report that once that value is in the `.env` your previous fixes / merge works well. 

```docker exec -it isle-apache-dev bash -c "ls -lha /var/www/html"
total 8.0K
drwxrwxr-x 2 islandora islandora 4.0K Nov  7 18:22 .
drwxr-xr-x 1 root      root      4.0K Oct 22 16:38 ..
```

**Comment by g7morris** [comment]
@br2490 If you can update the `.env` as recommended above, happy to close ticket otherwise. ;)

**Comment by br2490** [comment]
Let's discuss this on today's call 👍

**Comment by g7morris** [comment]
@br2490 I don't believe these changes made it into the release?

**Comment by bseeger** [comment]
Is this ticket still relevant?

**Comment by McFateM** [comment]
Not sure where the UID issue stands these days, but when the UID was at 1000 I had LOTS of local/development problems since that UID was already taken on each of my Macs, and by different usernames.  Setting a uniform UID of 10000 seemed to quell a lot of problems, but I'm not sure how the .env's define this currently.

**Comment by g7morris** [comment]
This issue has not been solved per se. Like @McFateM above I also use the UID /GUID of 10000 to avoid.  I believe this issue is highlighted to be part of the upcoming ISLE Phase II - Optimization Sprint.   The larger question is using any type of user beyond root in any or all of the containers. There were issues in the apache and fedora containers early in ISLE development to this effect which is why there are islandora and tomcat users in the apache and fedora containers respectively. Perhaps it is an idea @bseeger ISLE Maintainers could schedule a meeting to organize / discuss this type of prioritized bug hunting to present to the ICG steering committee for review?

**Comment by g7morris** [comment]
Closing as "fixed" from ISLE 1.5.5 release
